src/main/java/com/ecore/roles/service/MembershipsService.java

```diff
@@ -0,0 +1,14 @@
+package com.ecore.roles.service;
+
+import com.ecore.roles.exception.ResourceNotFoundException;
+import com.ecore.roles.model.Membership;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface MembershipsService {
+
+    Membership assignRoleToMembership(Membership membership) throws ResourceNotFoundException;
```

Comment:
It could be just called "assignRoleAndCreate". Just "assignRole", because it's inside of the membership service.